### PR TITLE
Bug 2086506: hypershift: respect statefulset when upgrading ovnk

### PR DESCRIPTION
**Pull request by zshi-redhat**
Signed-off-by: Zenghui Shi <zshi@redhat.com>

### CODE REVIEW

File: pkg/network/ovn_kubernetes.go [hunk 1]
```diff
@@ -1310,11 +1371,38 @@ func daemonSetProgressing(ds *appsv1.DaemonSet, allowHung bool) bool {
...
+	s := "progressing"
+	if !progressing {
+		s = "complete"
+	}
+	klog.V(2).Infof("statefulset %s/%s rollout %s; %d/%d scheduled; %d available; generation %d -> %d",
+		ss.Namespace, ss.Name, s, status.ReadyReplicas, status.Replicas,
+		status.AvailableReplicas, ss.Generation, status.ObservedGeneration)
+
+	if !progressing {
+		klog.V(2).Infof("daemonset %s/%s rollout complete", ss.Namespace, ss.Name)
```
Comment (squeed) [inline pkg/network/ovn_kubernetes.go:1393]: statefulset :)

File: pkg/bootstrap/types.go [hunk 2]
```diff
@@ -41,15 +40,24 @@ type OVNConfigBoostrapResult struct {
 	HyperShiftConfig *OVNHyperShiftBootstrapResult
 }
 
+type OVNUpdateStatus struct {
```
Comment (squeed) [inline pkg/bootstrap/types.go:43]: Can you add some kind of comment about what this is for? Something about "the status of a Deployment or StatefulSet"?

File: pkg/bootstrap/types.go [hunk 3]
```diff
@@ -41,15 +40,24 @@ type OVNConfigBoostrapResult struct {
...
-	ClusterInitiator        string
-	ExistingMasterDaemonset *appsv1.DaemonSet
-	ExistingNodeDaemonset   *appsv1.DaemonSet
-	ExistingIPsecDaemonset  *appsv1.DaemonSet
-	OVNKubernetesConfig     *OVNConfigBoostrapResult
-	PrePullerDaemonset      *appsv1.DaemonSet
-	FlowsConfig             *FlowsConfig
+	MasterAddresses       []string
+	ClusterInitiator      string
+	MasterUpdateStatus    *OVNUpdateStatus
```
Comment (squeed) [inline pkg/bootstrap/types.go:55]: Can you add some comments about what all of these are? In other words, which resources they correspond to?

**Comment by squeed** [comment]
A few small comments, but otherwise looks good! Thanks!
(Note that tests are failing)

File: pkg/network/ovn_kubernetes.go (hunk 1 above)
Comment (zshi-redhat) [reply pkg/network/ovn_kubernetes.go]: Fixed

**Comment by zshi-redhat** [comment]
> A few small comments, but otherwise looks good! Thanks! (Note that tests are failing)

fixed the tests.

**Comment by squeed** [comment]
/approve
will lgtm when hypershift tests go green

**Comment by squeed** [comment]
> > /approve will lgtm when hypershift tests go green
> 
> hypershift test failed for the same reason as explained in [#1441 (comment)](https://github.com/openshift/cluster-network-operator/pull/1441#issuecomment-1129127323)

Aha, makes sense
/lgtm

**Pull request merged**